src/App.tsx

```diff
@@ -1,46 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
+import { Tabs } from './components/Tabs/Tabs';
+import { Tab } from './components/Tabs/Types/Tab';
 
-export const tabs = [
+export const tabs: Tab[] = [
   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
 export const App: React.FC = () => {
-  return (
-    <div className="section">
-      <h1 className="title">Selected tab is Tab 1</h1>
-
-      <div data-cy="TabsComponent">
-        <div className="tabs is-boxed">
-          <ul>
-            <li className="is-active" data-cy="Tab">
-              <a href="#tab-1" data-cy="TabLink">
-                Tab 1
-              </a>
-            </li>
+  const [tab, setTab] = useState<Tab>(tabs[0]);
+  const findTab = (ta: Tab) => {
+    const foundTab = tabs.find(t => t.id === ta.id);
 
-            <li data-cy="Tab">
-              <a href="#tab-2" data-cy="TabLink">
-                Tab 2
-              </a>
-            </li>
+    if (foundTab) {
+      setTab(foundTab);
+    }
+  };
 
-            <li data-cy="Tab">
-              <a href="#tab-3" data-cy="TabLink">
-                Tab 3
-              </a>
-            </li>
-          </ul>
-        </div>
+  return (
+    <>
+      <div className="section">
+        <h1 className="title">Selected tab is {tab.title}</h1>
 
-        <div className="block" data-cy="TabContent">
-          Some text 1
-        </div>
+        <Tabs tabs={tabs} findTab={findTab} selectedTab={tab} />
```

Comment:
In tests you are expected to pass exact props, probably this adjustment should help. 

```
      mount(
        <Tabs
          tabs={tabs}
          selectedTabId="tab-20"
          onTabSelected={cy.spy().as('onTabSelected')}
        />
```